waiter/src/waiter/process_request.clj

```diff
@@ -52,6 +52,13 @@
            (org.eclipse.jetty.io EofException)
            (org.eclipse.jetty.server HttpChannel HttpOutput Response)))
 
+(defn make-auth-user-map
+  "Creates a map containing the username and principal from a request"
+  [{:keys [authorization/metadata authorization/principal]}]
+  ;; TODO creating this map is unnecessary, we can destructure from the request directly
```

Comment:
This seems like a small TODO.  Can we either do it or remove the comment if we feel like we should leave this function in place?

Reply:
Removed the TODO, do not want to introduce those changes in this PR.